This is an evaluation of the given dataset using python.


In [1]:
#Importing the necessary libraries
import pandas as pd
import seaborn as sns
import matplotlib
import matplotlib.pyplot as plt

In [2]:
#reading the csv
df=pd.read_csv("DVDA1Salesdata.csv")

In [3]:
df.head()

AccountId AccountName Region     Division       City State Tier Month  \
0       1116    Account1   West  DIAMONDBACK  PHOENIX W    AZ  Low   Aug   
1       1116    Account1   West  DIAMONDBACK  PHOENIX W    AZ  Low   Oct   
2       2391    Account2   East    MINUTEMEN   HARTFORD    CT  Med   Jun   
3       2391    Account2   East    MINUTEMEN   HARTFORD    CT  Med   Mar   
4       2397    Account3   East    MINUTEMEN  WORCESTER    MA  Med   Sep   

   Sales2015  Sales2016  Units2015  Units2016  TargetAchevied2015  \
0       0.00   13208.52        0.0        4.0                0.70   
1   10500.78   23114.91        3.0        7.0                0.84   
2       0.00    6627.00        0.0        3.0                1.15   
3   19881.00   13254.00        9.0        6.0                1.33   
4    3684.48       0.00        1.0        0.0                1.02   

   TargetAchevied2016  
0                1.53  
1                1.31  
2                1.29  
3                1.17  
4                1.53

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3709 entries, 0 to 3708
Data columns (total 14 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   AccountId           3709 non-null   int64  
 1   AccountName         3709 non-null   object 
 2   Region              3709 non-null   object 
 3   Division            3709 non-null   object 
 4   City                3709 non-null   object 
 5   State               3709 non-null   object 
 6   Tier                3709 non-null   object 
 7   Month               3709 non-null   object 
 8   Sales2015           3709 non-null   float64
 9   Sales2016           3709 non-null   float64
 10  Units2015           3709 non-null   float64
 11  Units2016           3709 non-null   float64
 12  TargetAchevied2015  3709 non-null   float64
 13  TargetAchevied2016  3709 non-null   float64
dtypes: float64(6), int64(1), object(7)
memory usage: 405.8+ KB


In [5]:
#Check for null values
df.isnull().sum()

AccountId             0
AccountName           0
Region                0
Division              0
City                  0
State                 0
Tier                  0
Month                 0
Sales2015             0
Sales2016             0
Units2015             0
Units2016             0
TargetAchevied2015    0
TargetAchevied2016    0
dtype: int64

In [5]:
#Quick Statistics on the numerical data
df.describe()

AccountId      Sales2015      Sales2016    Units2015   Units2016  \
count  3.709000e+03    3709.000000    3709.000000  3709.000000  3709.00000   
mean   7.716535e+04    6134.912371    7998.039527     2.293567     3.00791   
std    2.417615e+05    9195.070153   11130.551032     3.659052     4.31373   
min    1.116000e+03  -11053.440000  -59090.810000    -3.000000   -17.00000   
25%    3.774000e+03       0.000000    1886.690000     0.000000     1.00000   
50%    6.500000e+03    3128.340000    3918.770000     1.000000     1.83330   
75%    1.041900e+04    7837.540000   10122.400000     3.000000     4.00000   
max    1.219499e+06  145760.120000  168033.360000    53.000000    61.00000   

       TargetAchevied2015  TargetAchevied2016  
count         3709.000000         3709.000000  
mean             1.067396            1.242354  
std              0.211269            0.192727  
min              0.700000            0.910000  
25%              0.880000            1.080000  
50%              1.070000            1.240000  
75%              1.250000            1.410000  
max              1.420000            1.570000

Given the above information, it is found that the dataframe attributes Sales2015,Sales2016,Units2015,Units2016 have values lesser than 0.

These values will be removed as they are considered outliers.

In [9]:
"""
Removing the outliers from the data using
Sales data<0 and Units sold<0
"""
df_filtered=df[~((df["Sales2015"]<0)|(df["Sales2016"]<0)| (df["Units2015"]<0)| (df["Units2016"]<0))]

In [10]:
#Cross checking the information pertaining to the existing dataframe
df_filtered.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3674 entries, 0 to 3708
Data columns (total 14 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   AccountId           3674 non-null   int64  
 1   AccountName         3674 non-null   object 
 2   Region              3674 non-null   object 
 3   Division            3674 non-null   object 
 4   City                3674 non-null   object 
 5   State               3674 non-null   object 
 6   Tier                3674 non-null   object 
 7   Month               3674 non-null   object 
 8   Sales2015           3674 non-null   float64
 9   Sales2016           3674 non-null   float64
 10  Units2015           3674 non-null   float64
 11  Units2016           3674 non-null   float64
 12  TargetAchevied2015  3674 non-null   float64
 13  TargetAchevied2016  3674 non-null   float64
dtypes: float64(6), int64(1), object(7)
memory usage: 430.5+ KB


Given the above, it is observed that a total of 35 entries were removed when the above filters were applied.

(Originally, there were 3709 values, post filtering there are 3674 values)

In [11]:
#Quick statistics on the filtered dataframe
df_filtered.describe()

AccountId      Sales2015      Sales2016    Units2015    Units2016  \
count  3.674000e+03    3674.000000    3674.000000  3674.000000  3674.000000   
mean   7.601594e+04    6201.268861    8067.269041     2.318138     3.030939   
std    2.392690e+05    9205.813644   11069.986204     3.665375     4.300082   
min    1.116000e+03       0.000000       0.000000     0.000000     0.000000   
25%    3.773250e+03       0.000000    2161.370000     0.000000     1.000000   
50%    6.486000e+03    3128.340000    4095.690000     1.000000     2.000000   
75%    1.041075e+04    7893.930000   10122.400000     3.000000     4.000000   
max    1.219499e+06  145760.120000  168033.360000    53.000000    61.000000   

       TargetAchevied2015  TargetAchevied2016  
count         3674.000000         3674.000000  
mean             1.067025            1.242548  
std              0.211406            0.192727  
min              0.700000            0.910000  
25%              0.880000            1.080000  
50%              1.070000            1.240000  
75%              1.250000            1.410000  
max              1.420000            1.570000

In [12]:
#Exporting the filtered dataframe as a csv to import the same to tableau
df_filtered.to_csv('DVDA1_filtered.csv',index=False)